src/include/optimizer/query_to_operator_transformer.h

```diff
@@ -52,6 +53,7 @@ class QueryToOperatorTransformer : public SqlNodeVisitor {
   void Visit(const parser::UpdateStatement *op) override;
   void Visit(const parser::CopyStatement *op) override;
   void Visit(const parser::AnalyzeStatement *op) override;
+  concurrency::Transaction *txn = nullptr;
```

Comment:
1. Move to private.
2. Initialize in constructor.
3. Forward declare `Transaction`.